### ISSUE TITLE
Table Component: Problems with newColumn()

### ISSUE THREAD

**Issue by shinrichs** [Servoy/servoy-extra-components#32]
We have some problems with the newColumn() function in Table Component. If we just create a column with a dataprovider, it works, but if we create a column with e. g. styleClassDataprovider or valuelist, the two properties don't work correctly especially if we try to copy a column object for e. g.  column sorting. In case of column sorting we delete all actual columns and create them new while 'copying' the column objects.

We already fixed the problem with the styleClassDataprovider by changing the function newColumn() in table_server.js and function parameters in .spec file. So if we have a styleClassDataprovider we put the styleClassDataproviderId as parameter to newColumn() like normal dataprovider:

`$scope.api.newColumn = function(dataproviderid,index,styleClassDataproviderid) {
  if (!$scope.model.columns) $scope.model.columns = [];
    var insertPosition = (index == undefined) ? $scope.model.columns.length : ((index == -1 || index >      $scope.model.columns.length) ? $scope.model.columns.length : index);
	 for(var i = $scope.model.columns.length; i > insertPosition; i--) {
		  $scope.model.columns[i] = $scope.model.columns[i - 1]; 
	 }
	 if(styleClassDataproviderid){
		 $scope.model.columns[insertPosition] = {'dataprovider':dataproviderid,'styleClassDataprovider':styleClassDataproviderid}; 
	 } else{
		 $scope.model.columns[insertPosition] = {'dataprovider':dataproviderid}; 
	 }
	 return $scope.model.columns[insertPosition];
}`

That works but it don't work if we use valuelists in Table Component.

Improvement proposal: Is it possible to implement column sorting function, e. g. change the index of a column, set a visible flag etc.?

**Comment by shinrichs** [comment]
I found the problem with the valuelist. If I try to copy the valuelist property in column object into another column object I get a JSDataset in case the column object comes from runtime element and a JSValuelist object in case the column object comes from solution model. 

Now I'm using the name property from the JSValuelist object and put it as parameter into newColumn() function with dataprovider and optional styleclassDataProvider, that's working perfectly!

Whatever, I guess it's a good improvement for the component implementing column sorting as a standard function.

**Comment by jcompagner** [comment]
i still don't get this

we don't want to have an endless amount of properties in the "newColumn" api

why can't you just do:

newColumn("dataproviderid", index).styleClassDataproviderid = x 

?
thats the whole idea, that all the other properties can just be set on the return value of newColumn

**Comment by shinrichs** [comment]
Because it isn't  still working. 
If I try to set the styleClassDataprovider, for example the dataproviderId of a calculation that return any styleClass, AFTER initializing new column object like your solution, the styleClassDataprovider doesn't work anymore respectively doesn't render correctly. I guess we tried many solutions but don't get a working styleClassDataprovider, so we modified your version and set the styleClassDataprovider at newColumn(), that's still working.

As a helpful information, we use the version 1.1.4 of Servoy Extra Components...
